src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherContext.java

```diff
@@ -12,15 +15,26 @@
  */
 public class ExtendedEmailPublisherContext {
     private ExtendedEmailPublisher publisher;
-    private AbstractBuild<?, ?> build;
+    private Run<?, ?> run;
+    private FilePath workspace;
     private EmailTrigger trigger;
-    private BuildListener listener;
+    private TaskListener listener;
     private Launcher launcher;
     private Multimap<String, EmailTrigger> triggered;
-    
+
+    @Deprecated
     public ExtendedEmailPublisherContext(ExtendedEmailPublisher publisher, AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) {
         this.publisher = publisher;
-        this.build = build;
+        this.run = build;
+        this.workspace = build.getWorkspace();
+        this.launcher = launcher;
+        this.listener = listener;

```

Comment:
Wouldn't be better to directly call the new constructor? Just for easier maintenance in the future.